File: 24.07.2024/find_second_largest.py
def find_second_largest(arr):
    first, second = float('-inf'), float('-inf')
    for num in arr:
        if num > first:
            second = first
            first = num
        elif first > num > second:
            second = num
    return second

arr = [12, 35, 1, 10, 34, 1]
result = find_second_largest(arr)
print("Second largest element is", result if result != float('-inf') else "None")

File: 24.07.2024/findSecondLargest.c
#include <stdio.h>
int findSecondLargest(int arr[], int n) {
    int first = arr[0], second = -1;
    for (int i = 1; i < n; i++) {
        if (arr[i] > first) {
            second = first;
            first = arr[i];
        } else if (arr[i] > second && arr[i] != first) {
            second = arr[i];
        }
    }
    return second;
}

int main() {
    int arr[] = {12, 35, 1, 10, 34, 1};
    int n = sizeof(arr) / sizeof(arr[0]);
    int result = findSecondLargest(arr, n);
    if (result != -1)
        printf("Second largest element is %d\n", result);
    else
        printf("There is no second largest element\n");
    return 0;
}
